// console-table-printer/dist/src/utils/table-helpers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.cellText = exports.createHeaderAsRow = exports.renderTableHorizontalBorders = exports.findMaxLenOfColumn = exports.createRow = exports.createColum = exports.createTableHorizontalBorders = exports.textWithPadding = exports.convertRawRowOptionsToStanrd = void 0;
function convertRawRowOptionsToStanrd(options) {
    if (options) {
        return {
            color: options.color,
        };
    }
    return undefined;
}
exports.convertRawRowOptionsToStanrd = convertRawRowOptionsToStanrd;
function textWithPadding(text, alignment, mxColumnLen) {
    const curTextSize = text.length;
    switch (alignment) {
        case 'left':
            return text.padEnd(mxColumnLen);
        case 'center':
            return text
                .padStart((mxColumnLen - curTextSize) / 2 + curTextSize)
                .padEnd(mxColumnLen);
        case 'right':
        default:
            return text.padStart(mxColumnLen);
    }
}
exports.textWithPadding = textWithPadding;
function createTableHorizontalBorders({ left, mid, right, other, }, column_lengths) {
    // ╚
    let ret = left;
    // ╚═══════╩═══════════════════════════════════════╩════════╩
    column_lengths.forEach((len) => {
        ret += other.repeat(len + 2);
        ret += mid;
    });
    // ╚═══════╩═══════════════════════════════════════╩════════
    ret = ret.slice(0, -mid.length);
    // ╚═══════╩═══════════════════════════════════════╩════════╝
    ret += right;
    return ret;
}
exports.createTableHorizontalBorders = createTableHorizontalBorders;
function createColum(name) {
    return { name };
}
exports.createColum = createColum;
function createRow(color, text) {
    return { color, text };
}
exports.createRow = createRow;
function findMaxLenOfColumn(column, rows) {
    const column_name = column.name;
    let max_ln = `${column_name}`.length;
    rows.forEach((row) => {
        max_ln = Math.max(max_ln, `${row.text[column_name] || ''}`.length);
    });
    return max_ln;
}
exports.findMaxLenOfColumn = findMaxLenOfColumn;
function renderTableHorizontalBorders(style, column_lengths) {
    const str = createTableHorizontalBorders(style, column_lengths);
    return str;
}
exports.renderTableHorizontalBorders = renderTableHorizontalBorders;
function createHeaderAsRow(createRowFn, columns) {
    const headerBolor = 'white_bold';
    const row = createRowFn(headerBolor, {});
    columns.forEach((column) => {
        row.text[column.name] = column.name;
    });
    return row;
}
exports.createHeaderAsRow = createHeaderAsRow;
function cellText(text) {
    return text === undefined || text === null ? '' : text;
}
exports.cellText = cellText;
